robot_calibration/src/capture_features.cpp

```diff
@@ -0,0 +1,304 @@
+// Author: Jack Center
+
+#include <fstream>
+#include <ctime>
+
+#include <ros/ros.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <rosbag/query.h>
+
+#include <std_msgs/String.h>
+#include <robot_calibration_msgs/CalibrationData.h>
+#include <robot_calibration_msgs/CaptureConfig.h>
+
+#include <robot_calibration/capture/chain_manager.h>
+#include <robot_calibration/capture/feature_finder_loader.h>
+
+#include <camera_calibration_parsers/parse.h>
+#include <robot_calibration/ceres/optimizer.h>
+#include <robot_calibration/camera_info.h>
+#include <robot_calibration/load_bag.h>
+
+#include <boost/foreach.hpp>  // for rosbag iterator
+
+#include <robot_calibration/capture_features.h>
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv,"robot_calibration");
+  ros::NodeHandle nh("~");
+
+  if (!check_parameters(nh))
+    return -1;
+
+  // Set up the calibration
+// TODO: make timeout a define
+  robot_calibration::ChainManager chain_manager(nh, 0.1);     // Manages kinematic chains
+
+  robot_calibration::FeatureFinderMap finders;               // Holds the available feature finders  
+  if (!get_feature_finders(nh, finders)) 
+    return -1;    
```

Comment:
Same comment here about logging an error before returning.  Also, if you have multiple failure codes from the main you should return different values from each.  In this case, I would just return -2 for this one.